third_party/dependency_analyzer/src/test/io/bazel/rulesscala/dependencyanalyzer3/UnusedDependencyCheckerTest.scala

```diff
@@ -0,0 +1,64 @@
+package io.bazel.rulesscala.dependencyanalyzer
+
+import org.scalatest.funsuite._
+import io.bazel.rulesscala.utils.Scala3CompilerUtils
+import io.bazel.rulesscala.utils.Scala3CompilerUtils._
+
+class UnusedDependencyCheckerTest extends AnyFunSuite {
+  def compileWithUnusedDependencyChecker(code: String, withDirect: List[(String, String)] = Nil): CompileResult = {
+    val extraClasspath = withDirect.map(_._1)
+
+    Scala3CompilerUtils.runCompiler(
+      code = code,
+      extraClasspath = extraClasspath,
+      dependencyAnalyzerParamsOpt =
+        Some(
+          DependencyAnalyzerTestParams(
+            directJars = withDirect.map(_._1),
+            directTargets = withDirect.map(_._2),
+            unusedDeps = true,
+            dependencyTrackingMethod = DependencyTrackingMethod.HighLevel
+          )
+        )
+    )
+  }
+
+  test("error on unused direct dependencies") {
+    val testCode =
+      """object Foo {
+        |}
+      """.stripMargin
+
+    val commonsTarget = "//commons:Target"
+
+    val direct = List(apacheCommonsClasspath -> encodeLabel(commonsTarget))
+    val result = compileWithUnusedDependencyChecker(testCode, withDirect = direct)
+
+    result match {
+      case Success() => fail("Expected compilation to fail")
+      case Failure(errorMessages) => assert(errorMessages.exists { msg =>
+        msg.contains(commonsTarget) &&
+          msg.contains(s"buildozer 'remove deps $commonsTarget' $defaultTarget")
+      })
+    }
+
+  }
+
+  ignore("do not error on used direct dependencies") {
```

Comment:
Why is this test ignored? If it can't be fixed now, it would be very useful to have explanatory comment